backend/back4app-cloud-code/back4app-cloud-code/checkAnswer.js

```diff
@@ -0,0 +1,111 @@
+class CheckAnswerEngine {
+
+  // checks the correctness of a given answer by comparing to a list of acceptable answers
+  static checkAnswerList(givenAnswer, acceptableAnswers, blockAnswers) {
+    
+    const answerChecks = acceptableAnswers.map((ans) => (this.checkAnswer(givenAnswer, ans)));
+    if(answerChecks.includes("correct,final")){
+      return {isCorrect: true, isFinal: true}
+    }
+    if(answerChecks.includes("correct")){
+      return {isCorrect: true, isFinal: false}
+    }
+    const blockChecks = blockAnswers.map((ans) => (this.checkAnswer(givenAnswer, ans)));
+    if(blockChecks.includes("correct,final")){
+      return {isCorrect: false, isFinal: true}
+    }
+    return {isCorrect: false, isFinal: false}
+  }
+
+  // replaces number words with the correct number, to standardize answer format
+  static replaceNumber(word){
```

Comment:
still wondering how'd you'd support numbers outside 1-16. how can we make sure we have extensive coverage without making it repetitive?